bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/StringListType.java

```diff
@@ -38,12 +38,16 @@ public StringListType() {
     }
 
     public StringListType(StringType... rows) {
-        typeDetails = new ArrayList<String>(rows.length);
+        typeDetails = new ArrayList<>(rows.length);
         for (StringType row : rows) {
```

Comment:
not part of your PR but this could well be rewritten to use streams: `Arrays.stream(rows).map(r -> r.toString()).collect(Collectors.asList())` and feed this into your c`tor. wdyt?